client/app/pages/queries/QuerySource.jsx

```diff
@@ -225,6 +230,31 @@ function QuerySource(props) {
     [query]
   );
 
+  const editSchedule = useCallback(() => {
+    const canScheduleQuery = true; // TODO: Use real value
+    if (!query.can_edit || !canScheduleQuery) {
+      return;
+    }
+
+    const intervals = clientConfig.queryRefreshIntervals;
+    const allowedIntervals = policy.getQueryRefreshIntervals();
+    const refreshOptions = isArray(allowedIntervals) ? intersection(intervals, allowedIntervals) : intervals;
+
+    ScheduleDialog.showModal({
+      schedule: query.schedule,
+      refreshOptions,
+    }).result.then(schedule => {
+      updateQuery(query, { schedule }).then(setQuery);
+    });
+  }, [query]);
+
+  const updateQueryDescription = useCallback(
+    description => {
+      updateQuery(query, { description }).then(setQuery);
+    },
+    [query]
+  );
```

Comment:
WDYT of having this as an action in utils:
https://github.com/getredash/redash/blob/7914602a3c5ac07ecebd1ca00aa70727414da692/client/app/pages/queries/utils/index.js#L32-L35
?

(Actually the Schedule Dialog thing could also be moved there, since it's the same logic for the Query View)

Reply:
https://github.com/getredash/redash/pull/4476/commits/b9de64e475e89b114c8facded18394bd9057d34d